WordPress/Classes/Utility/Universal Links/Migration/WPAdminConvertibleRouter.swift

```diff
@@ -0,0 +1,95 @@
+/// A router that handles routes that can be converted to /wp-admin links.
+///
+/// Note that this is a workaround for an infinite redirect issue between WordPress and Jetpack
+/// when both apps are installed.
+///
+/// This can be removed once we remove the Universal Link routes for the WordPress app.
+struct WPAdminConvertibleRouter: LinkRouter {
```

Comment:
To clarify, the idea of this workaround is:
- Where possible, construct wp-admin links which **do not redirect** to mobile apps
- If wp-admin links are not possible, do nothing in the app

And this solution wouldn't be viable if wp-admin pages would actually redirect to mobile apps?

Reply:
> this solution wouldn't be viable if wp-admin pages would actually redirect to mobile apps?

Yes, you're correct 💯.  I was taking advantage of the fact that we aren't handling any wp-admin routes in iOS. Of course, this is only temporary.

My thinking was that the conversion to a wp-admin link, for some qualified links, allows some limited UX mitigation where users would still have their intent fulfilled in Safari; instead of being redirected into the app without navigating anywhere.

Reply:
Thanks for clarifying! 

Given that we recommend not using the apps together, I think this solution even exceeds expectations. Avoiding the loops might be even enough.

If we would actually want to have WordPress and Jetpack apps seamlessly work together, then we would need to look for approaches that support all the cases.